units/src/amount.rs

```diff
@@ -1476,8 +1484,17 @@ impl fmt::Debug for SignedAmount {
 // Just using Bitcoin denominated string.
 impl fmt::Display for SignedAmount {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        self.fmt_value_in(f, Denomination::Bitcoin)?;
-        write!(f, " {}", Denomination::Bitcoin)
+        let mut format_options = FormatOptions::default();
+        format_options.precision.replace(f.precision().unwrap_or(8));
+
+        fmt_satoshi_in(
+            self.unsigned_abs().to_sat(),
+            self.is_negative(),
+            f,
+            Denomination::Bitcoin,
+            false,
+            format_options,
+        )
```

Comment:
I would have expected to see the same code as in `impl fmt::Display for Amount`.

Said another way, there probably should be a test for `SignedAmount` that is equivalent to `trailing_zeros_for_amount`. (I see an unrelated bug there too, you can discover and fix that if you like :)

Reply:
Now the code uses the formatting options passed to `f: &mut fmt::Formatter` . 

As for the bug any hint :)

Reply:
`SignedAmount` isn't printing "BTC" like `Amount` is - it is unrelated to this PR though.

Reply:
Lol, I didn't see that either. I assumed the bug was related to the weird `if` .. `if else` tree that didn't have a final `else` clause.

This whole logic is pretty gnarly. I have a fuzztest that I am using locally that we should add:

```
use std::str::FromStr;

use honggfuzz::fuzz;

use bitcoin::Amount;

fn do_test(data: &[u8]) {
    if data.len() < 3 { return }

    let prec = data[0] as usize % 10; 
    let width = data[1] as usize % 20; 

    let data_str = String::from_utf8_lossy(&data[2..]);
    let amt = match Amount::from_str(&data_str) {
        Ok(amt) => amt,
        Err(_) => return,
    };  

    let amt2 = amt.to_btc();

    println!("Prec: {prec}");
    println!("Width: {width}");
    println!("str: {data_str}");
    println!("amt2: {amt2}");
    assert_eq!(
        format!("{amt:width$.prec$}"),
        format!("{amt2:width$.prec$} BTC"),
    );  
}

fn main() {
    loop {
        fuzz!(|data| {
            do_test(data);
        }); 
    }   
}
```

That I would suggest adding to this PR. (Once it starts working, please remove the printfs, tack a copyright header on, and add the boilerplate testcase that exists in the other test cases.)

It would also be OK if you dropped the check for consistent `width` behavior (which needs to be tweaked anyway I think to add a `- 4` because `Amount` outputs the suffix ` BTC` while `f64` does not, and this changes how many digits `width` implies), in order to keep this PR reasonably scoped.

But eventually we *should* have such a check, and maybe for other formatter flags as well.